src/distilabel/pipeline/base.py

```diff
@@ -255,6 +264,9 @@ def _cache_location(self) -> CacheLocation:
 
     def _cache(self) -> None:
         """Saves the `BasePipeline` using the `_cache_filename`."""
+        if not self._use_cache:
+            return
```

Comment:
I think we can still cache the pipeline execution even if `use_cache` is false. The parameter is more about using the cache or not when running the pipeline.